lib/identity_cache/query_api.rb

```diff
@@ -246,96 +272,102 @@ def find_batch(ids)
         ids.map{ |id| records_by_id[id] }
       end
 
-      def prefetch_associations(associations, records)
-        associations = hashify_includes_structure(associations)
+      def fetch_embedded_associations(records)
+        associations = embedded_associations
+        return true if associations.empty?
 
-        associations.each do |association, sub_associations|
-          case
-          when details = cached_has_manys[association]
+        return false unless primary_cache_index_enabled
 
-            if details[:embed] == true
-              child_records = records.map(&details[:cached_accessor_name].to_sym).flatten
+        records_by_id = records.index_by(&:id)
+        cached_records = fetch_multi(records.map(&:id))
```

Comment:
The whole point of this method *is* to go the cache for the associations.  However, those associations are embedded on a record, so we need to fetch the record to get at the associations.

I don't know what you mean about *incorrect associations*.  In relation to a mutation, the cache would have been invalidated to avoid fetching anything incorrect.